corehq/apps/userreports/models.py

```diff
@@ -460,20 +464,31 @@ def get_doc_id(cls, domain, table_id):
         return '{}{}-{}'.format(cls._datasource_id_prefix, domain, table_id)
 
     @classmethod
-    def all(cls):
+    @skippable_quickcache([], skip_arg='rebuild')
```

Comment:
yea, having the map in the first place and caching it so we don't have to regenerate it for each call